Add git-crypt for secrets management and implement nullmailer with Mailjet SMTP

teleboas opened the pull request:
This PR introduces two main improvements to our infrastructure management:

Git-crypt Integration

- Added git-crypt configuration for secure secrets management
- Created comprehensive documentation for GPG key generation and user onboarding
- Updated `.gitattributes` to encrypt sensitive files (secrets.yml, certificates, SSH keys)


Email Infrastructure

- Implemented nullmailer role for handling system emails
- Added Mailjet SMTP configuration for reliable email delivery
- Updated `jobs/configure.yml` playbook to include mail configuration

Documentation has been updated to reflect new pre-requisites and setup procedures for repository access.
The changes improve security by encrypting sensitive files while providing a robust email delivery system for system notifications.

teleboas committed:
Add 1 git-crypt collaborator

New collaborators:

	F554B0DE Vincent Bataille <vince@bu.gs>

teleboas committed:
Add mail routing by Mailjet through nullmailer

teleboas committed:
Add mailjet encrypted credentials

alexgarel commented on group_vars/all/mails.yml:
```diff
@@ -0,0 +1,27 @@
+
+# All mails to <user>@localhost or <user>@<nullmailer__mailname>
+# are redirected to this email (ie. no local mailbox)
+nullmailer__admin: infra@openfoodfacts.org
```
why not root@openfoodfacts.org ?

teleboas replied:
I'd love it if we could use an address other than root for those emails. E-mails addressed to root@<default_domain> often flags that some services might not be set up correctly. This small change will help us easily distinguish between mail sent by servers that are properly configured and relay via MailJet and mail sent by our legacy servers still using their own MTAs.

alexgarel replied:
ok, good idea.

alexgarel commented on group_vars/all/mails.yml:
```diff
@@ -0,0 +1,27 @@
+
+# Configure to nullmailer to use host_domain as default mailing domain
+# so unconfigured mail are emited in @<host_domain>
+nullmailer__domain: '{{ host_domain }}'
```
why not openfoodfacts.org ?

If we use other domains we will have to configure DKIM and all the things for those subdomains…

teleboas replied:
For most of our servers, {{ host_domain }} will indeed be openfoodfacts.org (or infra.openfoodfacts.org if we stick with this naming scheme). We'll need to configure SPF and DKIM for all the domains we plan to use, so it might be better to standardize and use a few consistent ones. However, I wasn't entirely sure if we should limit ourselves to just one domain (openfoodfacts.org).

Couldn't we consider the possibility of using other domains, such as openfoodfacts.net for some pre-production servers for instance?

alexgarel replied:
Normally servers are on openfoodfacts.org, but yes we can discuss it in the future.

alexgarel reviewed (approved):
Great, I just have some minor remarks.

I sent you my gpg key, because I'm eager to test !

teleboas committed:
Add 1 git-crypt collaborator

New collaborators:

	45F741B7 Alex Garel (Open Food Facts) <alex@openfoodfacts.org>

teleboas committed:
chore: update examples in the documentation
